editor/contexts/Auth.tsx

```diff
@@ -18,6 +20,20 @@ export default function AuthProvider({
     (typeof window !== "undefined" && window.location.origin + BASE_PATH) ||
     undefined;
 
+  const router = useRouter();
+
+  /**
+   * Callback function that routes the user to their original destination
+   * after login
+   */
+  const onRedirectCallback = useCallback(
+    (appState: AppState | undefined) => {
+      const returnTo = appState?.returnTo || window.location.pathname;
+      router.push(returnTo);
+    },
+    [router]
+  );
```

Comment:
As you can see below, we were already setting the `{ returnTo: "some-url"}` data when logging in, but we did not have a handler configured to do anything about it. Now the app properly routes to the `returnTo` destination 👍